sim/pokemon.js

```diff
@@ -1083,12 +1085,13 @@ class Pokemon {
 	 * @param {Pokemon?} source
 	 * @param {Effect?} effect
 	 */
-	damage(d, source = null, effect = null) {
+	damage(d, source = null, effect = null, direct = false) {
 		if (!this.hp) return 0;
 		if (d < 1 && d > 0) d = 1;
 		d = Math.floor(d);
 		if (isNaN(d)) return 0;
 		if (d <= 0) return 0;
+		if (!direct) this.hurt = true;
```

Comment:
Also I would probably just put this is `battle.damage`, so the `direct` flag would remain unnecessary.